lib/portlayer/storage/vsphere/toolbox_common.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package vsphere
+
+import (
+	"fmt"
+	"path"
+
+	"github.com/vmware/govmomi/guest"
+	"github.com/vmware/govmomi/guest/toolbox"
+	"github.com/vmware/govmomi/vim25/types"
+	"github.com/vmware/vic/lib/archive"
+	"github.com/vmware/vic/pkg/trace"
+	"github.com/vmware/vic/pkg/vsphere/vm"
+)
+
+const (
+	DiskLabelQueryName   = "disk-label"
+	FilterSpecQueryName  = "filter-spec"
+	SkipRecurseQueryName = "skip-recurse"
+	SkipDataQueryName    = "skip-data"
+)
+
+// Parse Archive builds an archive url with disklabel, filtersec, recursive, and data booleans.
+func BuildArchiveURL(op trace.Operation, disklabel, target string, fs *archive.FilterSpec, recurse, data bool) (string, error) {
+	encodedSpec, err := archive.EncodeFilterSpec(op, fs)
+	if err != nil {
+		return "", err
+	}
+	target = path.Join("/archive:/", target)
+
+	// if diskLabel is longer than 16 characters, then the function was passed a containerID
+	// use containerfs as the diskLabel
+	if len(disklabel) > 16 {
+		disklabel = "containerfs"
+	}
+
+	// note that the query parameters a SkipX for recurse and data so values are inverted
+	target += fmt.Sprintf("?%s=%s&%s=%s&%s=%t&%s=%t", DiskLabelQueryName, disklabel, FilterSpecQueryName, *encodedSpec, SkipRecurseQueryName, !recurse, SkipDataQueryName, !data)
```

Comment:
This ought to be:
```go
	target += (url.Values{
		DiskLabelQueryName: []string{diskLabel},
                FilterSpecQueryName: []string{encodedSpec},
                ...
	}).Encode()
```

This way any special chars within the values will be properly url encoded.